src/Cpp/1-getting-started/1-HelloWindow/Application.cpp

```diff
@@ -0,0 +1,94 @@
+#include "Application.hpp"
+#include "Input/Input.hpp"
+
+#include <GLFW/glfw3.h>
+
+Application::Application(const std::string_view title)
+    : _input(nullptr), _window(nullptr), _width(0), _height(0), _title(title)
+{
+}
+
+Application::~Application()
+{
+    Application::Cleanup();
+}
+
+void Application::Run()
+{
+    if (!Initialize())
+    {
+        return;
+    }
+
+    while (!glfwWindowShouldClose(_window))
+    {
+        _input->Update(
+            static_cast<float>(_width) / 2.0f,
+            static_cast<float>(_height) / 2.0f);
+        glfwPollEvents();
+        Update();
+        Render();
+        glfwSwapBuffers(_window);
+    }
+}
+
+// ReSharper disable once CppMemberFunctionMayBeConst
+void Application::Cleanup()
+{
+    glfwTerminate();
+}
+
+// ReSharper disable once CppMemberFunctionMayBeConst
+void Application::Close()
+{
+    glfwSetWindowShouldClose(_window, GLFW_TRUE);
+}
+
+bool Application::Initialize()
+{
+    if (!glfwInit())
+    {
+        return false;
+    }
+
+    const auto primaryMonitor = glfwGetPrimaryMonitor();
+    const auto videoMode = glfwGetVideoMode(primaryMonitor);
+    _width = static_cast<std::int32_t>(videoMode->width * 0.9f);
+    _height = static_cast<std::int32_t>(videoMode->height * 0.9f);
+
+    glfwWindowHint(GLFW_SCALE_TO_MONITOR, GLFW_FALSE);
+    glfwWindowHint(GLFW_CLIENT_API, GLFW_NO_API);
+    _window = glfwCreateWindow(_width, _height, _title.data(), nullptr, nullptr);
+    if (_window == nullptr)
+    {
+        Cleanup();
+        return false;
+    }
+    _input = std::make_unique<Input>(_window);
+
+    const auto windowLeft = videoMode->width / 2 - _width / 2;
```

Comment:
avoid `auto` for primitive types, error sensitive regarding implicit conversions

Reply:
We resolved by vote see [here](https://github.com/GraphicsProgramming/learnd3d11/discussions/14)